src/System.Management.Automation/engine/Subsystem/Commands/GetSubsystemCommand.cs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#nullable enable
+
+namespace System.Management.Automation.Subsystem
+{
+    /// <summary>
+    /// Implementation of 'Get-Subsystem' cmdlet.
+    /// </summary>
+    [Experimental("PSSubsystemPluginModel", ExperimentAction.Show)]
+    [Cmdlet(VerbsCommon.Get, "Subsystem", DefaultParameterSetName = AllSet)]
+    [OutputType(typeof(SubsystemInfo))]
+    public sealed class GetSubsystemCommand : PSCmdlet
+    {
+        private const string AllSet = "GetAllSet";
+        private const string TypeSet = "GetByTypeSet";
+        private const string KindSet = "GetByKindSet";
+
+        /// <summary>
+        /// Gets or sets a concrete subsystem kind.
+        /// </summary>
+        [Parameter(Mandatory = true, ParameterSetName = KindSet, ValueFromPipeline = true)]
+        public SubsystemKind Kind { get; set; }
+
+        /// <summary>
+        /// Gets or sets the interface or abstract class type of a concrete subsystem.
+        /// </summary>
+        [Parameter(Mandatory = true, ParameterSetName = TypeSet, ValueFromPipeline = true)]
+        public Type? SubsystemType { get; set; }
```

Comment:
The compiler asks this property to be initialized with an notnull value if make it no-nullable. I don't it makes sense to initialize it with anything other than `null`.

Reply:
> The compiler asks this property to be initialized with an notnull value if make it no-nullable. I don't it makes sense to initialize it with anything other than `null`.

I've taken to initializing with ` = null!` but that's probably worse since you'd ideally keep the warnings in a constructor or `Dispose` method 😕. I wish there was a attribute for `NotNullFrom(string methodName)` or something.